File: python/luhn/luhn.py
class Luhn:
    def __init__(self, card_num):
        self.card_num = card_num.replace(" ","")
    def valid(self):
        def resdigit(dig):
            if 2*dig < 9:
               return (2*dig)
            else:
               return (2*dig-9)
        if len(self.card_num) > 1 and self.card_num.isnumeric():
           digsum = sum([int(e) for e in self.card_num[::-2]])
           digsum += sum([resdigit(int(o)) for o in self.card_num[-2::-2]])
           return digsum % 10 == 0
        else:
           return False

File: julia/luhn/luhn.jl
value(digit) = digit -'0'
double(digit) = digit < 9 ? digit : digit - 9

function luhn(cardNumber)
	digits = filter(!isspace, cardNumber) |> reverse
	length(digits) > 1 && all([isnumeric(d) for d in digits]) || return false
	even = sum(double(value(d) * 2) for d in digits[2:2:end])
	odd =sum(value(d) for d in digits[1:2:end])
	(even + odd) % 10 == 0
end
